src/run_common.cpp

```diff
@@ -36,6 +36,12 @@ void setup_common_run_options(
             "t is the current logical time and d is the amount of logical "
             "time that has passed since the start of the simulation.  "
             "t and d are floating-point numbers while n is an integer.")
+        ("worker-threads",
+            boost::program_options::value<int>()->value_name("worker-threads")->implicit_value(-1),
```

Comment:
I just remembered that there is a difference between `default_value`, which is used for a couple of options further up, and `implicit_value`, which is used here:

* `default_value` is used when the `--worker-threads` option is not given by the user at all.
* `implicit_value` is used when the user specifies `--worker-threads` but does not give a number (as opposed to e.g. `--worker-threads=3`).

From the discussion elsewhere in this PR, it looks to me like we want to keep the "old behaviour", which is multithreading by default. In that case, we should use `default_value`. But if we want to switch to single-threading by default, then `implicit_value` is the way to go.

Reply:
[Source](https://www.boost.org/doc/libs/release/doc/html/boost/program_options/typed_value.html#id-1_3_32_9_10_1_1_1_5_1-bb)

Reply:
well spotted :)